src/odin/maneuversbuilder.cc

```diff
@@ -2336,7 +2336,7 @@ bool ManeuversBuilder::IsTurnChannelManeuverCombinable(std::list<Maneuver>::iter
                                                        bool start_man) const {
 
   // Current maneuver must be a turn channel and not equal to the next maneuver
-  if (curr_man->turn_channel() && (curr_man != next_man)) {
+  if (curr_man->turn_channel() && (curr_man != next_man) && !next_man->IsDestinationType()) {
```

Comment:
Should we leave a TODOD note that we need to extend this logic (of not combining with the destination maneuver) to other places as well?

Are there tests which exercise this code-path when roundabout_exits is true & false?

Reply:
I'm actually not sure where it makes sense to extend this logic, as it seems to have been deliberately left out of the other collapse cases. I'll follow up with @dgearhart about that as soon as I can.

Re: tests, I mentioned in the OP that I'll follow up with a gurka test. Meanwhile @gknisely helped run RAD tests with this change and saw no diffs, so I'm fairly confident the fix is localized to the situation that triggers the bug.

Reply:
Okay. RAD showing 0 diffs is a good sign! :) 👍